File: backend/app/schemas/media.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class MediaFileOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    file_path: str
    file_type: str
    file_size: int
    mime_type: str
    uploaded_by: int
    created_at: datetime


class PaginatedMediaFiles(BaseModel):
    items: list[MediaFileOut]
    total: int
    page: int
    page_size: int

File: frontend/src/api/media.ts
import api from './client'

export interface MediaFile {
  id: number
  title: string
  file_path: string
  file_type: 'image' | 'video' | 'document'
  file_size: number
  mime_type: string
  uploaded_by: number
  created_at: string
}

export interface PaginatedMediaFiles {
  items: MediaFile[]
  total: number
  page: number
  page_size: number
}

export const mediaApi = {
  list: (page = 1, pageSize = 20) =>
    api.get<PaginatedMediaFiles>('/media', { params: { page, page_size: pageSize } }),

  upload: (file: File) => {
    const form = new FormData()
    form.append('file', file)
    return api.post<MediaFile>('/media/upload', form, {
      headers: { 'Content-Type': 'multipart/form-data' },
    })
  },

  delete: (id: number) =>
    api.delete(`/media/${id}`),
}
